dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/client/CommonDataSourceClient.java

```diff
@@ -101,6 +117,140 @@ public void checkClient() {
         }
     }
 
+    @Override
+    public List<String> getDatabaseList(String databasePattern) {
+        return this.jdbcTemplate.queryForList(getDatabaseListSql(databasePattern), String.class);
+    }
+
+    protected String getDatabaseListSql(String databasePattern) {
+        throw new UnsupportedOperationException("NOT_SUPPORT");
+    }
+
+    @Override
+    public List<String> getTableList(String dbName, String schemaName, String tablePattern) {
+        return this.jdbcTemplate.queryForList(getTableListSql(dbName, schemaName, tablePattern), String.class);
+    }
+
+    protected String getTableListSql(String dbName, String schemaName, String tablePattern) {
+        throw new UnsupportedOperationException("NOT_SUPPORT");
+    }
+
+    @Override
+    public List<Map<String, Object>> getTableStruct(String dbName, String schemaName, String tableName) {
+        throw new UnsupportedOperationException("NOT_SUPPORT");
+    }
+
+    @Override
+    public MutableTriple<Map<String, String>, List<Map<String, Object>>, List<Map<String, String>>> executeSql(
```

Comment:
Can we return a pojo here?

Reply:
There will be other definitions for pojo